WordPress/gutenberg
# Popover: refactor to TypeScript

## What?


Part of #35744
Part of #42770

Refactor the `Popover` component to TypeScript.

Unit tests will be converted separately, so that they can offer a better guarantee that the changes in this PR won't be introducing regressions.

## Why?


See #35744 

## How?


Following the step-by-step guide in the [CONTRIBUTING guidelines](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CONTRIBUTING.md#refactoring-a-component-to-typescript)

Reviewing commit-by-commit may also offer an easier way to review this PR.

## Testing Instructions





- Project builds, all tests pass
- For Popover and its derived components (Autocomplete, BorderControl, BorderBoxControl, Dropdown, DropdownMenu, PaletteEdit, GradientPicker, Tooltip..), check Storybook and make sure that all examples behave like on `trunk`
- Do an extensive regression test in the post and site editor
